ArduCopter/mode_plancktracking.cpp

```diff
@@ -53,6 +53,35 @@ void ModePlanckTracking::run() {
             relative_angle);
     }
 
+    //Check for tether high tension
+    bool high_tension = copter.planck_interface.is_tether_high_tension() || copter.planck_interface.is_tether_timed_out();
+    bool use_positive_throttle = high_tension;
+    if(high_tension) {
+        if(!copter.planck_interface.check_if_high_tension_failed(copter.current_loc.alt)) { //High tension hasn't failed
+            //While in high tension:
+            // - If actively tracking the tag, continue to do so, but use pos throttle
+            // - If not tracking the tag, but tracking the ground GPS, continue to do so, but use pos throttle
+            // - If not tracking tag or ground GPS, use zero velocity command (if gps available, zero attitude if not) with pos throttle
+            if(!copter.planck_interface.get_tag_tracking_state() && !copter.planck_interface.get_commbox_state()) {
+                if(copter.position_ok()) {
+                  copter.planck_interface.override_with_zero_vel_cmd();
+                } else {
+                  copter.planck_interface.override_with_zero_att_cmd();
+                }
+            }
+        } else { //High tension has failed
+            //If high tension has failed, attempt to use a planck land or regular land
+            if(copter.planck_interface.ready_for_land()) {
+                copter.set_mode_planck_RTB_or_planck_land(ModeReason::GCS_FAILSAFE);
+                use_positive_throttle = false; //Use normal altitude control
+            } else {
+                copter.set_mode_land_with_pause(ModeReason::GCS_FAILSAFE);
+                copter.mode_land.run();
```

Comment:
Why do we need to call mode land.run()? Won't it change modes and run automatically?

Reply:
Not until the next loop. This `run()` call isn't strictly necessary, but if we don't call any `run()` methods then it means no controller outputs are generated for 1 loop.

Reply:
Ah, I see